projects/plugins/jetpack/_inc/client/components/jetpack-notices/index.jsx

```diff
@@ -175,7 +177,61 @@ UserUnlinked.propTypes = {
 class JetpackNotices extends React.Component {
 	static displayName = 'JetpackNotices';
 
+	constructor( props ) {
+		super( props );
+
+		const cookieParsed = cookie.parse( document.cookie );
+		const dismissedNotices = Object.keys( cookieParsed )
+			.filter( key => key.startsWith( 'jetpack_deprecate_dismissed' ) )
+			.reduce( ( acc, key ) => {
+				acc[ key.replace( 'jetpack_deprecate_dismissed[', '' ).replace( /]/g, '' ) ] = '1';
+				return acc;
+			}, {} );
+
+		this.state = {
+			dismissedNotices,
+		};
+	}
+
+	componentDidMount() {
+		const noticeHandlers = {};
+		const noticeKeys = []; // Add more notice keys here
+
+		noticeKeys.forEach( noticeKey => {
+			noticeHandlers[ noticeKey ] = this.dismissNotice.bind( this, noticeKey );
+		} );
+
+		this.setState( { noticeHandlers } );
+	}
+
+	dismissNotice = noticeKey => {
+		this.setState( prevState => ( {
+			dismissedNotices: {
+				...prevState.dismissedNotices,
+				[ noticeKey ]: '1',
+			},
+		} ) );
+
+		document.cookie = `jetpack_deprecate_dismissed[${ noticeKey }]=1; expires=Fri, 31 Dec 9999 23:59:59 GMT; path=/; Secure; SameSite=None`;
+	};
+
+	isNoticeDismissed = noticeKey => {
+		return this.state.dismissedNotices[ noticeKey ] === '1';
+	};
+
 	render() {
+		// Add notices here. Example:
+		// const notices = [
```

Comment:
I wonder if we could only define the notices once in `deprecate.php` and not have to add them in 2 places.. eg add them in the initial state?

Reply:
Do you mean, add them in `class-deprecate.php`? I could add the notice info as an variable on the `jetpack-deprecate` script (within `enqueue_admin_scripts`), is this along the lines of what you were thinking?

Reply:
I've found this works well - and I was able to remove the JS logic to check if the notice is dismissed as well, simplifying things further. 14dbb7da610c8cd694262b3e66460928d0bc46bb